site/docs/references/mixin-syntax.md

```diff
@@ -1,124 +1,253 @@
 * Layouts - easily describe complex layouts
 * Effects - easily describe complex effects
-* Macros - use JS to define the CSS macros you need
+* Macros - use code to define the CSS macros you need
+
+Mixins can be applied using either CSS or JavaScript, and can recive parameters.
+
+>**Note**  
+> If you need to return only a single declaration value using code, we recommend **Stylable** [formatters](./formatters.md). 
+
+## CSS mixins
 
-## Example usage
+Any CSS stylesheet, class or element that is defined in a **Stylable** CSS file can be used as a mixin source. You can use either a local class or element, or import the mixin from a different stylesheet.
 
-The value `textTooltip` of the external file `my-mixins` is imported. The class selector `.submitButton` uses the mixin syntax and applies parameters. In this case, to wait `300` milliseconds to display the `dataTooltip` hover text on the button. 
+In the following example, a locally defined class is used as a mixin in the same stylesheet.
 
 ```css
 /* CSS */
-:import {
-    -st-from: "./my-mixins";
-    -st-named: textTooltip;
-}
-.submitButton {
-    -st-mixin: textTooltip(300, dataTooltip); /* apply mixin */
+.style-mixin {
+    color: green;
+    background: yellow;
 }
+.someClass {
+    -st-mixin: style-mixin;     
+}  
+```
+```css
+/* CSS output */
+...
+.someClass {
+    color: green; /* from local mixin */
+    background: yellow; /* from local mixin */
+}  
 ```
 
-## Defining mixins in CSS
+Here is an example of a **Stylable** CSS file that is imported and mixed into the  classes of a different stylesheet. The `.rootMixedIn` class as a stylesheet and `classMixedIn` as a class.
 
-Every CSS class defined in a **stylable** CSS file can be used as a mixin.
+```css
+/* CSS mixin file - mixins.st.css */
+.root {
+    color: purple;
+}
 
-``` css
+.someClass {
+    color: green;
+}
+```
 
-/* file a.st.css */
-.a{
-    color:red;
+``` css
+/* CSS file - example.st.css - imports the above mixin */
+:import {
+    -st-from: './mixins.st.css';
+    -st-default: MyComp;
+    -st-named: someClass;
 }
 
-/* file b.st.css */
-:import{
-    -st-from:'./a.st.css';
-    -st-named:a;
+.rootMixedIn {
+    -st-mixin: MyComp; /* stylesheet mixin */
 }
-.b{
-    -st-mixin:a;
+
+.classMixedIn {
+    -st-mixin: someClass; /* class mixin */
 }
 ```
 
-``` css
+```css
 /* CSS output */
-.b{
-    color:red;
+.rootMixedIn {
+    color: purple; /* from stylesheet mixin */
 }
-```
 
-## Defining mixins in JavaScript or TypeScript
+.rootMixedIn .someClass { /* ruleset added as a result of the stylesheet mixin */
+    color: green;
+}
 
-You can easily create new mixins using JavaScript or TypeScript.
+.classMixedIn {
+    color: green; /* from class mixin */
+}
+```
 
-## Applying multiple mixins
+### CSS mixin with parameters and variables
 
-You can apply multiple mixins in the same line.
+CSS mixins can accept named parameters in the following format:
+ `mixin(variableName valueOverride)`. 
```

Comment:
add a second parameter to the syntax example:

`mixin(variableName valueOverride, variableName2 valueOverride2)`

It might be also nice to show here that it works in multiline aswell:

```css
mixin(
    variableName valueOverride, 
    variableName2 valueOverride2
)
````